flake: TestTxClientTestSuite/TestMultiConnBroadcast port binding retry reuses same privval gRPC port

## Summary

`TestTxClientTestSuite/TestMultiConnBroadcast/All_Fail` failed on `main` due to a port binding conflict on the privval gRPC listener. The retry logic in `testnode.NewNetworkWithRetry` does **not** reset `PrivValidatorGRPCListenAddr`, so all 5 retries re-use the same port that originally failed.

## CI

- Post-merge CI run: https://github.com/celestiaorg/celestia-app/actions/runs/24666582764/job/72125730971
- Commit: `c1f6e5c0ed1d6724a982327e07acbe2eeaf75468` (on `main`)

## Failure output

```
--- FAIL: TestTxClientTestSuite (46.48s)
    --- FAIL: TestTxClientTestSuite/TestMultiConnBroadcast (26.15s)
        --- FAIL: TestTxClientTestSuite/TestMultiConnBroadcast/All_Fail (15.50s)
            network.go:23: port binding error on attempt 1/5, retrying after 1s: failed to listen for privval gRPC: listen tcp 127.0.0.1:52843: bind: address already in use
            network.go:23: port binding error on attempt 2/5, retrying after 2s: failed to listen for privval gRPC: listen tcp 127.0.0.1:52843: bind: address already in use
            network.go:23: port binding error on attempt 3/5, retrying after 3s: failed to listen for privval gRPC: listen tcp 127.0.0.1:52843: bind: address already in use
            network.go:23: port binding error on attempt 4/5, retrying after 4s: failed to listen for privval gRPC: listen tcp 127.0.0.1:52843: bind: address already in use
            network.go:23: port binding error on attempt 5/5, retrying after 5s: failed to listen for privval gRPC: listen tcp 127.0.0.1:52843: bind: address already in use
            network.go:23: Failed to start network after 5 attempts
FAIL
FAIL	github.com/celestiaorg/celestia-app/v9/pkg/user	279.337s
```

## Root cause

`test/util/testnode/config.go:172` configures `tmCfg.PrivValidatorGRPCListenAddr` with a free port:

```go
tmCfg.PrivValidatorGRPCListenAddr = fmt.Sprintf("127.0.0.1:%d", MustGetFreePort())
```

But the retry block in `test/util/testnode/network.go:36-46` only re-rolls the RPC, P2P, gRPC, App gRPC, and App API addresses:

```go
if isPortBindingError(err) {
    t.Logf("port binding error on attempt %d/%d, retrying after %ds: %v", attempt+1, maxRetries, attempt+1, err)
    time.Sleep(time.Duration(attempt+1) * time.Second)
    config.TmConfig.RPC.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
    config.TmConfig.P2P.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
    config.TmConfig.RPC.GRPCListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
    config.AppConfig.GRPC.Address = fmt.Sprintf("127.0.0.1:%d", MustGetFreePort())
    config.AppConfig.API.Enable = true
    config.AppConfig.API.Address = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
    continue
}
```

`PrivValidatorGRPCListenAddr` is missing, so when the original port collides, every retry fails with the same error.

## Proposed fix

Add the missing line to the retry block:

```go
config.TmConfig.PrivValidatorGRPCListenAddr = fmt.Sprintf("127.0.0.1:%d", MustGetFreePort())
```

--- a/test/util/testnode/network.go
+++ b/test/util/testnode/network.go
@@ -36,12 +36,7 @@ func NewNetworkWithRetry(t testing.TB, config *Config, maxRetries int) (cctx Con
 			if isPortBindingError(err) {
 				t.Logf("port binding error on attempt %d/%d, retrying after %ds: %v", attempt+1, maxRetries, attempt+1, err)
 				time.Sleep(time.Duration(attempt+1) * time.Second)
-				config.TmConfig.RPC.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
-				config.TmConfig.P2P.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
-				config.TmConfig.RPC.GRPCListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
-				config.AppConfig.GRPC.Address = fmt.Sprintf("127.0.0.1:%d", MustGetFreePort())
-				config.AppConfig.API.Enable = true
-				config.AppConfig.API.Address = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
+				reassignListenPorts(config)
 				continue
 			}
 			t.Fatalf("Failed to start network after %d attempts: %v", attempt+1, err)
@@ -122,6 +117,20 @@ func tryStartNetwork(t testing.TB, config *Config) (cctx Context, rpcAddr, grpcA
 	}
 
 	return cctx, config.TmConfig.RPC.ListenAddress, config.AppConfig.GRPC.Address, cleanup, nil
+}
+
+// reassignListenPorts reassigns every listener address in config to a new free
+// port. It must cover every port-bearing field set in DefaultTendermintConfig
+// and DefaultAppConfig; missing any one of them makes the retry loop in
+// NewNetworkWithRetry re-use a stale port on every attempt (see #7137).
+func reassignListenPorts(config *Config) {
+	config.TmConfig.RPC.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
+	config.TmConfig.P2P.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
+	config.TmConfig.RPC.GRPCListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
+	config.TmConfig.PrivValidatorGRPCListenAddr = fmt.Sprintf("127.0.0.1:%d", MustGetFreePort())
+	config.AppConfig.GRPC.Address = fmt.Sprintf("127.0.0.1:%d", MustGetFreePort())
+	config.AppConfig.API.Enable = true
+	config.AppConfig.API.Address = fmt.Sprintf("tcp://127.0.0.1:%d", MustGetFreePort())
 }
 
 // isPortBindingError checks if an error is related to port binding failures

--- a/test/util/testnode/network_test.go
+++ b/test/util/testnode/network_test.go
@@ -1,0 +1,41 @@
+package testnode
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestReassignListenPorts exercises every listen-port field that the retry
+// path in NewNetworkWithRetry must re-roll after a port binding error. If any
+// of these fields is missed, a transient collision on that port will recur on
+// every retry and the test will fail after maxRetries attempts (see #7137).
+func TestReassignListenPorts(t *testing.T) {
+	cfg := DefaultConfig()
+
+	before := struct {
+		tmRPC     string
+		tmP2P     string
+		tmRPCGRPC string
+		tmPrivVal string
+		appGRPC   string
+		appAPI    string
+	}{
+		tmRPC:     cfg.TmConfig.RPC.ListenAddress,
+		tmP2P:     cfg.TmConfig.P2P.ListenAddress,
+		tmRPCGRPC: cfg.TmConfig.RPC.GRPCListenAddress,
+		tmPrivVal: cfg.TmConfig.PrivValidatorGRPCListenAddr,
+		appGRPC:   cfg.AppConfig.GRPC.Address,
+		appAPI:    cfg.AppConfig.API.Address,
+	}
+
+	reassignListenPorts(cfg)
+
+	assert.NotEqual(t, before.tmRPC, cfg.TmConfig.RPC.ListenAddress, "TmConfig.RPC.ListenAddress should be reassigned")
+	assert.NotEqual(t, before.tmP2P, cfg.TmConfig.P2P.ListenAddress, "TmConfig.P2P.ListenAddress should be reassigned")
+	assert.NotEqual(t, before.tmRPCGRPC, cfg.TmConfig.RPC.GRPCListenAddress, "TmConfig.RPC.GRPCListenAddress should be reassigned")
+	assert.NotEqual(t, before.tmPrivVal, cfg.TmConfig.PrivValidatorGRPCListenAddr, "TmConfig.PrivValidatorGRPCListenAddr should be reassigned")
+	assert.NotEqual(t, before.appGRPC, cfg.AppConfig.GRPC.Address, "AppConfig.GRPC.Address should be reassigned")
+	assert.NotEqual(t, before.appAPI, cfg.AppConfig.API.Address, "AppConfig.API.Address should be reassigned")
+	assert.True(t, cfg.AppConfig.API.Enable, "AppConfig.API.Enable should remain true")
+}

